service-worker.tmpl

```diff
@@ -47,141 +45,108 @@ var IgnoreUrlParametersMatching = [<%= ignoreUrlParametersMatching %>];
 var <%- functionName %> = <%= externalFunctions[functionName] %>;
 <% }); %>
 
-var mappings = populateCurrentCacheNames(PrecacheConfig, CacheNamePrefix, self.location);
-var AbsoluteUrlToCacheName = mappings.absoluteUrlToCacheName;
-var CurrentCacheNamesToAbsoluteUrl = mappings.currentCacheNamesToAbsoluteUrl;
-
-function deleteAllCaches() {
-  return caches.keys().then(function(cacheNames) {
-    return Promise.all(
-      cacheNames.map(function(cacheName) {
-        return caches.delete(cacheName);
-      })
-    );
+var HashParamName = '_sw-precache';
+var UrlsToCacheKeys = new Map(
+  PrecacheConfig.map(function(item) {
+    var relativeUrl = item[0];
+    var hash = item[1];
+    var absoluteUrl = new URL(relativeUrl, self.location);
+    var cacheKey = createCacheKey(absoluteUrl, HashParamName, hash);
+    return [absoluteUrl.toString(), cacheKey];
+  })
+);
+
+function setOfCachedUrls(cache) {
```

Comment:
I'd prefer to keep things consistent throughout, and move to ES2015 _en masse_. https://github.com/GoogleChrome/sw-precache/issues/82